modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1956,15 +1935,13 @@ private CompletableFuture<WriteIntentSwitchReplicatedInfo> applyWriteIntentSwitc
                 indexIdsAtRwTxBeginTs(transactionId)
         );
 
-        CompletableFuture<Object> resultFuture = new CompletableFuture<>();
-
-        applyCmdWithRetryOnSafeTimeReorderException(wiSwitchCmd, resultFuture);
-
-        return resultFuture
+        return applyCmdWithExceptionHandling(wiSwitchCmd)
                 .exceptionally(e -> {
                     LOG.warn("Failed to complete transaction cleanup command [txId=" + transactionId + ']', e);
 
-                    return nullCompletedFuture();
+                    ExceptionUtils.sneakyThrow(e);
```

Comment:
If the exception is rethrown anyway, we don't need `exceptionally()`. But why is this changed to throwing the exception?

Reply:
Probably because the exception is lost if not re-thrown here?

Reply:
My question was probably confusing, I'll break it down.

1. The code was specifically using `exceptionally()` to make sure an exception doesn't propagate. I'm not sure this is the correct thing to do here, but we could ask the author. @denis-chudov could you please explain why the exception was just being logged and its propagation stopped?
2. If we actually don't need to stop exception propagation, then this `exceptionally()` stop is not needed. If we just want to log it (and let it propagate), then it seems more natural to use `whenComplete((res, ex) -> { if (ex != null) log(ex) }) instead of `exceptionally()`

Reply:
It seems that this code is a result of accumulated error due to multiple changes. At some moment, we did async write intent switch and didn't care about the result of the future: that's why `exceptionally` with logging was made. After that, it was moved to return sentence. It's about time to fix it (which @ascherbakoff just did): the exception should be propagated in order to make a retry in `WriteIntentSwitchProcessor#switchWriteIntentsWithRetry`